src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -414,4 +434,18 @@ protected function hasValidValues(array $values, $type = null): bool
 
         return true;
     }
+
+    /**
+     * Check if an uuid is valid or not.
+     *
+     * @see regexp taken from https://github.com/ramsey/uuid
+     *
+     * @param string|int $identifier
+     *
+     * @return bool
+     */
+    protected function isValidUuid($identifier): bool
+    {
+        return (bool) preg_match('/^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}$/', $identifier);
+    }
```

Comment:
I'm -1 for this. I'd just let the failing doctrine operation throw.

Reply:
Yeah, I agree. Also because you can create your own uuid generator, why would you validate that here?

Reply:
Then we should not support uuid in here, why are we testing that an int is a valid int then ?

Reply:
Not exactly sure what you mean but to me it should work like this: Whatever I set to be the identifier in doctrine should be searchable and whatever the user is querying on our API, it should not validate the value of the identifier. If doctrine can find an entity based on it it's fine, otherwise not. I don't see why we should validate the value passed by the customer here. Maybe you can elaborate?

Reply:
The thing is, when passing a non-valid uuid (but according to the type-checking it's valid because it's a valid string) the filter throws an errors 500. 

IMHO checking that you are passing a valid guid when the doctrine metadatas says that should be a guid is the right call, with this, if the guid is not valid, then it will try to search on it, which will lead to the SearchFilter, not filtering which is the expected behaviour.

AFAIK, removing this breaks how we expect filters to work.

Reply:
I have my own id generator:

```
    /**
     * @ORM\Id
     * @ORM\Column(name="voucher", type="string", length=9, options={"fixed":true})
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class="App\Doctrine\ORM\EntityManager\VoucherCodeGenerator")
     */
    private $voucher;
```

It's a 9 char, specially generated string. I could not search on that because your UUID validation would fail :(

Reply:
no, you can, because it's not guid type.
see https://github.com/api-platform/core/pull/1844/files#diff-77ec54980605d6062780fcd93ba31c64R274

Reply:
My believe is that if someone wants a search by uuid he should do it's own because our search filter doesn't support it yet.

As we discussed before, the search filter needs a complete refactor so that it validates most of the needs and it's really hard to do without breaking the so called filter. 
I'd clearly deprecate this class and set up one or two new filters, for example:
- FieldSearchFilter (handles LIKE and such where clauses)
- IdentifierSearchFilter (handles strict equalities, IRIs and everything related to identifiers (`foo.bar=id`). This one would reuse the identifier denormalization introduced a few weeks back).